feat: improved update messaging on app bar server info

bwees opened the pull request:
## Description
Improves update dialog to highlight warning messages in yellow and also show them conditionally similar to the warning on the user icon

## Light Mode
<img width="369" height="576" alt="Screenshot 2025-10-19 at 10 01 18 PM" src="https://github.com/user-attachments/assets/13802932-0144-4e5e-8adc-c80c5ad099b6" />

<img width="367" height="578" alt="Screenshot 2025-10-19 at 10 00 51 PM" src="https://github.com/user-attachments/assets/2c6230de-4453-4661-858f-d2901b0c1d48" />

<img width="368" height="574" alt="Screenshot 2025-10-19 at 10 01 56 PM" src="https://github.com/user-attachments/assets/d1a6f048-b382-4185-9b9a-73959de97108" />

## Dark Mode

<img width="368" height="583" alt="Screenshot 2025-10-19 at 10 01 26 PM" src="https://github.com/user-attachments/assets/8a5875e1-4aa2-44ea-a954-a02b77b6866d" />

<img width="368" height="574" alt="Screenshot 2025-10-19 at 10 00 15 PM" src="https://github.com/user-attachments/assets/c5958445-13ee-4810-a68d-87c211b4a720" />

<img width="367" height="574" alt="Screenshot 2025-10-19 at 10 01 47 PM" src="https://github.com/user-attachments/assets/0ba36b88-2bd3-4dcf-b40e-d6a3ace3efc6" />

bwees committed:
feat: improved update messaging on app bar server info

bwees committed:
chore: message improvements

bwees committed:
chore: failed to fetch version error message

bwees committed:
feat: open latest release when tapping "Update" on server out of date message

bwees committed:
fix: text alignment states

jrasm91 commented on i18n/en.json:
```diff
@@ -1547,13 +1546,12 @@
-  "profile_drawer_server_out_of_date_major": "Server is out of date. Please update to the latest major version.",
-  "profile_drawer_server_out_of_date_minor": "Server is out of date. Please update to the latest minor version.",
+  "profile_drawer_server_out_of_date": "A server update is available.",
+  "profile_drawer_unable_to_check_version": "Unable to check app or server version",
```
I wonder if we should have a way to mark this as needing to be translated again, since it'll keep being the old value for non-en languages.

alextran1502 replied:
I think just removing these old keys and using new keys with simplified key names like `out_of_date_message`...etc

jrasm91 reviewed (approved):
LGTM

bwees committed:
chore: code review updates

bwees committed:
Apply suggestion from @alextran1502

Co-authored-by: Alex <alex.tran1502@gmail.com>

bwees committed:
Apply suggestion from @alextran1502

Co-authored-by: Alex <alex.tran1502@gmail.com>

alextran1502 commented on mobile/lib/widgets/common/app_bar_dialog/app_bar_server_info.dart:
```diff
@@ -45,17 +78,45 @@ class AppBarServerInfo extends HookConsumerWidget {
+                        Expanded(
+                          child: Text(
+                            serverInfoState.versionMismatchErrorMessage,
+                            textAlign: (serverInfoState.isClientOutOfDate || serverInfoState.isServerOutOfDate)
```
Do we need to have a condition for this? maybe using the same style is ok?

alextran1502 commented on mobile/lib/widgets/common/app_bar_dialog/app_bar_server_info.dart:
```diff
@@ -45,17 +78,45 @@ class AppBarServerInfo extends HookConsumerWidget {
+                            overflow: TextOverflow.ellipsis,
+                          ),
+                        ),
+                        if (serverInfoState.isClientOutOfDate || serverInfoState.isServerOutOfDate)
```
So the conditions are already in `showWarning`, can you make a local variable to make the condition easier to understand for this specific case?

bwees committed:
chore: lots of rework of the version checking code to be cleaner

Added a semver utility class to simplify comparisons, broke the update notification logic into own widget, reworked view construction and colors.

bwees committed:
fix: show warnign without having to tap on app bar icon

alextran1502 committed:
chore: colors
